bitcoin/src/blockdata/witness.rs

```diff
@@ -133,18 +170,17 @@ impl Witness {
     /// Creates [`Witness`] object from an array of byte-arrays
     pub fn from_vec(vec: Vec<Vec<u8>>) -> Self {
         let witness_elements = vec.len();
+        let index_size = witness_elements * U32_SIZE;
 
         let content_size: usize = vec
             .iter()
             .map(|el| el.len() + VarInt(el.len() as u64).len())
             .sum();
-        let mut content = vec![0u8; content_size];
+        let mut content = vec![0u8; content_size + index_size];
```

Comment:
Oh, this was wasteful before and "still is". We know the exact content length and not using `std::io::Read` so the reasons for zeroing are not valid here. However because adding indices at the end requires `unsafe` zeroing became kinda justified. Maybe `VecDequeue` would work here better than in the other places but still not sure.